### ISSUE TITLE
Jam incremental build not working properly

### ISSUE THREAD

**Issue by dbear496** [lincity-ng/lincity-ng#67]
Jam seems to be rebuilding most files even when they are current, so builds are taking way longer than they need to.

To reproduce, run `jam`, and then run `jam` again. Even when no files are changed, the second invocation rebuilds most files.

**Comment by dbear496** [comment]
I have a suspicion that the issue may stem from the config.h file because when running `jam -dc`, the first line is `newer config.h`. Lots of files include `config.h`, so if Jam thinks it is out of date, then it will rebuild all those files that include it.

**Comment by dbear496** [comment]
I have found that this is related to the following warning I get during build:
```
ar: `u' modifier ignored since `D' is the default (see `U')
```

I run ArchLinux, and evidently, the binutils from the Arch repo is configured with `--enable-deterministic-archives` which inserts zero for all archive member timestamps.

**Comment by dbear496** [comment]
I sent an email to the FT-Jam devs, and they said the project is no longer maintained. I also doubt the original Jam is maintained either, since there does not appear to be any recent update and the source code repository is unavailable. It may be prudent to switch to a different build tool that is actively maintained.

**Comment by dbear496** [comment]
This is fixed in dbear496@d587a1626e40ae477205829e579fe2a741a27af9 by switching to cmake. This commit depends on some of my PRs so those would have to be merged first before merging this commit into the main project.
